### ISSUE TITLE
GCP pubsub subscription format deprecation

### ISSUE THREAD

**Issue by marccardinal** [keel-hq/keel#133]
According to this page: https://cloud.google.com/container-registry/docs/configuring-notifications

```Note: Previously, the qualified URI was formatted as projects/my-project/topics/us.gcr.io%2Fmy-project. As of September 18, 2017, this is deprecated.```

It looks like keel is trying to create a subscription based on the deprecated format.

**Comment by rusenask** [comment]
Hi, thank you. It's interesting why is this still working for me. I will update subscription.

**Comment by marccardinal** [comment]
Oh so it works for you?

I opened a ticket with GCP support because I tried both ways:

* creating a topic called `projects/myproject-registry/topics/gcr`
* using a topic created by keel I believe `projects/myproject-registry/topics/us.gcr.io%2Fmyproject-registry`

I subscribed to both of the topics and made a piece of code in python to listen to both of them but receive none of the messages I'm supposed to get when I push new images to the registry.

**Comment by rusenask** [comment]
Yes, I am using Keel with GCP pubsub from the day it was created. Maybe existing subscriptions still get connected to GCR? Even today I had an update to my backend that was triggered by the pubsub, so it seems that only new subscriptions are affected. 

I know that you probably already checked, but do you have pubsub enabled for your kubernetes node pool?

**Comment by marccardinal** [comment]
Yes I added the scope to a new pool and deleted the default one.

I imagine that it would still be good to fix this since the deprecation message is clear about it and will let you know what the issue was after GCP support gets back to me... since I reproduced the no-message issue with a separate piece of code I don't think it is keel specific.

**Comment by marccardinal** [comment]
Ok so I found out that the user or serviceAccount used to push to GCR needs at least `roles/storage.admin` as per the documentation to be able to write. To have the pubsub notification messages generated that same user also needs the `roles/pubsub.publisher` role apparently.

After adding the latter I started receiving events on two topics:

```
projects/myproject-registry/topics/gcr
projects/myproject-registry/topics/us.gcr.io%2Fmyproject-registry
```

So I imagine that the second scheme will be eventually deprecated but that in the meantime both are working.

**Comment by rusenask** [comment]
Got an email today from gcloud. Will update this.

**Comment by rusenask** [comment]
fixed in https://github.com/keel-hq/keel/pull/161, will close this once a new image will be released.

**Comment by rusenask** [comment]
Available from `0.8.0`.
